### PR TITLE
feat(21552): make UITable's combobox more generic

### PR DESCRIPTION
**Pull request by oliversberzs**
Changed UITable's boolean select into a more generic combobox

### CODE REVIEW

**Review by tom-sap — APPROVED** [review]

File: .changeset/selfish-gorillas-confess.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@sap-ux/ui-components": major
+"@sap-ux/types": minor
```
Comment (devinea) [inline .changeset/selfish-gorillas-confess.md:3]: `@sap-ux/types` is not changed here? `types/package.json`

For `@sap-ux/ui-components` Wouldn't patch / minor be more appropriate?
Is it technically a breaking change?
`UIBooleanSelect` is only used in data editor as far as I am aware.

Comment (oliversberzs) [reply .changeset/selfish-gorillas-confess.md:3]: `@sap-ux/types` was changed cause both `UIColumn` and `ColumnControlType` was changed, added a optional field.

Yes, it is only used in the data editor, but shouldn't that then still be marked as a breaking change, cause previous api has been removed and this is an open source project?

Comment (oliversberzs) [reply .changeset/selfish-gorillas-confess.md]: @Klaus-Keller @devinea removed the change marker from `@sap-ux/types` and made the other changes a minor bump by adding support for both control types

**Review by devinea — APPROVED** [review]
Thanks for the updates @oliversberzs

**Review by tom-sap — APPROVED** [review]

**Pull request merged**